crates/event-streaming/proto/walrus/event/v1alpha/service.proto

```diff
@@ -0,0 +1,164 @@
+// along with cryptographic proofs that tie them to a specific checkpoint.
+// It is designed to be forward-compatible, allowing for proof mechanisms
+// to be upgraded over time without breaking clients.
+service EventStreamingService {
+    // StreamEvents opens a server-side stream that sends sets of events
+    // as they are finalized in checkpoints. Each message in the stream contains
+    // all relevant events for a single checkpoint, plus the necessary data to verify them.
+    rpc StreamEvents(StreamEventsRequest) returns (stream VerifiableEvents);
+
+    // GetNonInclusionProof allows a client to request a cryptographic proof
+    // that a specific event did NOT occur within a given range of checkpoints.
+    // This is a standard unary RPC call.
+    rpc GetNonInclusionProof(GetNonInclusionProofRequest) returns (GetNonInclusionProofResponse);
+}
+
+
+// ============== Event Streaming RPC Messages ==============
+
+// An enum defining the available proof types.
+enum ProofType {
+    // Defaults to the server's preferred/most modern proof type.
+    PROOF_TYPE_UNSPECIFIED = 0;
+
+    // The client does not require a proof.
+    PROOF_TYPE_NONE = 1;
+
+    // The V1 checkpoint contents proof.
+    PROOF_TYPE_CHECKPOINT_CONTENTS_V1 = 2;
+
+    // The V2 Merkle inclusion proof.
+    PROOF_TYPE_MERKLE_PROOF_V2 = 3;
+}
+
+message StreamEventsRequest {
+    // The checkpoint sequence number from which to start streaming.
+    // Events from checkpoints before this number will not be sent.
+    // Defaults to 0, starting from the first available checkpoint.
+    optional uint64 starting_checkpoint_sequence_number = 1;
+
+    // The method used to identify the event stream. The client MUST specify
+    // exactly one of these options. This makes the API explicit and avoids
+    // ambiguity between V1 (package-based) and V2 (object-based) streams.
+    oneof stream_selector {
+        // V1: Subscribe to all events emitted by a specific package.
+        PackageSubscription package_id = 2;
+
+        // V2: Subscribe to events managed by a dedicated stream head object.
+        StreamHead stream_head = 3;
+    }
+
+    // The desired proof types, in order of client preference.
+    // The server will attempt to provide the first type in this list that it
+    // supports for a given checkpoint. If empty, it uses its default behavior.
+    // Example: [MERKLE_PROOF_V2, CHECKPOINT_CONTENTS_V1] means client prefer V2,
+    // but will accept V1 if V2 is not available (would not accept response without
+    // a proof)
+    repeated ProofType desired_proof_types = 4;
+}
+
+// VerifiableEvents is the message type for the stream.
+// Each message represents a single checkpoint that contained Walrus events.
+message VerifiableEvents {
+    // The summary of the checkpoint containing these events. The client uses
+    // the `digest` field from this summary to verify the accompanying proof.
+    optional sui.rpc.v2beta.CheckpointSummary summary = 1;
```

Comment:
I suspect it makes more sense for the summary to be part of the proof, e.g., in v1 you would have a redundancy of summary (both here and in the contents).

Reply:
contents don't have the summary (afaik)